tt_metal/impl/debug/dprint_server.cpp

```diff
@@ -269,12 +270,16 @@ void DebugPrintServerContext::WaitForPrintsFinished() {
     // or whether any cores are waiting for a signal to be raised.
     // TODO(dma): once we have access to the device is there a way we can poll the device to
     // check whether more print data is coming?
+    size_t num_harts_waiting = 0;
     do {
         // No need to await if the server was killed already due to a hang.
         if (server_killed_due_to_hang_)
             break;
-        std::this_thread::sleep_for(std::chrono::milliseconds(20));
-    } while (hart_waiting_on_signal_.size() > 0 || new_data_last_iter_);
+        std::this_thread::sleep_for(std::chrono::milliseconds(50));
+        raise_wait_lock_.lock();
```

Comment:
FYI, I've personally found this more convenient and less error-prone to use https://en.cppreference.com/w/cpp/thread/lock_guard

Reply:
Good idea, maybe next round of touching this code I'll switch all the locks here to use this instead. I'll just merge this one for now